### PR TITLE
Click to deploy ui: remove ipName and make hostName optional

### PR DESCRIPTION
**Pull request by lluunn**
/cc @kunmingg 
/cc @abhi-g

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1449)

### CODE REVIEW

File: components/gcp-click-to-deploy/src/DeployForm.tsx [hunk 1]
```diff
@@ -319,16 +321,22 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
...
         return;
       }
     }
 
+    if (this.state.hostName === '') {
+      this.setState({
+        hostName: this.state.deploymentName + '.endpoints.' + this.state.project + '.cloud.goog',
+      });
+    }
+
```
Comment (abhi-g) [inline components/gcp-click-to-deploy/src/DeployForm.tsx]: Does this mean that if a user wants they can specify a completely different hostname?

Comment (kunmingg) [reply components/gcp-click-to-deploy/src/DeployForm.tsx]: Yes, currently we only support the default one.
We can add support to custom hostname later.

Comment (kunmingg) [reply components/gcp-click-to-deploy/src/DeployForm.tsx]: We can remove hostname field from UI and print it in logs?

**Comment by lluunn** [comment]
Removed hostname field from UI, PTAL

**Comment by lluunn** [comment]
screen shot 2018-08-31 at 11 01 41

**Pull request merged**